src/cart/src/cart/crt_iv.c

```diff
@@ -2191,8 +2195,9 @@ crt_ivsync_rpc_issue(struct crt_ivns_internal *ivns_internal, uint32_t class_id,
 	if (iv_sync_cb == NULL)
 		D_GOTO(exit, rc = -DER_NOMEM);
 
-	iv_sync_cb->isc_sync_type = sync_type;
-
+	iv_sync_cb->isc_sync_type = *sync_type;
+	sync_type->ivs_comp_cb = NULL;
+	sync_type->ivs_comp_cb_arg = NULL;
```

Comment:
we shouldnt modify incoming argument 'sync_type'

Reply:
Hmm, this will stop the ivs_comp_cb will be missed or called multiple times, basically save a lot hassle for error handling case.

Reply:
I am not sure why/how this saves in error handling cases. If this somehow stops ivs_comp_cb() to be called twice then more reasons to not hide it, as it would indicate some error in logic somewhere, and more things than double completion could be in trouble.


This can also be called from handle_ivupdate_response() where it is part of allocated iv_info.

Reply:
It basically means if the callback is being taken care in crt_ivsync_rpc_issue(), indicated by reset the callback to NULL, then it does not need to execute the callback in crt_iv_update_internal().

Otherwise, it will check a lot cases if the callback has been executed especially during failure cases.